kratos/tests/cpp_tests/geometries/test_nurbs_volume.cpp

```diff
@@ -0,0 +1,457 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Manuel Messmer
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "testing/testing.h"
+#include "containers/pointer_vector.h"
+#include "geometries/nurbs_volume_geometry.h"
+#include "tests/cpp_tests/geometries/test_geometry.h"
+#include "geometries/point.h"
+
+namespace Kratos {
+namespace Testing {
+
+    typedef Node<3> NodeType;
+
+    NurbsVolumeGeometry<PointerVector<NodeType>> GenerateTruncatedPyramid() {
+        // Construct Truncated Pyramid with: lower_base = 2x2; uper_base = 1.8x1.8; heigth = 4
+        PointerVector<NodeType> points;
+        double t = 1.0;
+        std::vector<double> z_direction = {0, 2.0/3.0, 4.0/3.0, 2.0, 8.0/3.0, 10/3.0, 4.0};
+        std::vector<double> y_direction = {-1.0, -1.0/3.0, 1.0/3.0, 1.0};
+        std::vector<double> x_direction = {-1.0, -2.0/3.0, -1.0/3.0, 0.0, 1.0/3.0, 2.0/3.0, 1.0};
+        std::size_t id = 1;
+        for( auto i : z_direction){
+            for( double j : y_direction) {
+                for( auto k : x_direction) {
+                    double x = k*t;
+                    double y = j*t;
+                    double z = i;
+                    points.push_back(NodeType::Pointer(new NodeType(id, x, y, z)));
```

Comment:
```suggestion
                    points.push_back(Kratos::make_shared<NodeType>(id, x, y, z));
```

Reply:
you can compute the number of points. So you can actually get rid of the push back and insert it at the correct position.

Reply:
Done.